src/dev/eslint/run_eslint_custom.ts

```diff
@@ -0,0 +1,303 @@
+      );
+    }
+
+    const usedConcurrency = Math.max(
+      concurrency > availableParallelism ? availableParallelism : concurrency
+    );
+
+    log.info(`Using eslint config at ${eslintConf}.`);
+
+    if (formatter) {
+      log.info(`Using results formatter at ${formatter}.`);
+    }
+
+    log.info(`Running with ${usedConcurrency} workers.`);
+
+    const items = getPackages(REPO_ROOT);
+
+    const thunks: Array<() => Promise<EslintRunResult>> = items
+      .filter((item) =>
+        filter ? item.directory.includes(filter) || item.name.includes(filter) : true
+      )
+      .map((pkg) => {
+        return () => {
+          if (!formatter) {
+            log.info(`Linting ${pkg.directory} (config: ${eslintConf})`);
+          }
+          return runEslintOnEntry(pkg, {
+            json: !!formatter,
+            stream: !formatter,
+            eslintConf,
+          }).then((res) => {
+            if (!formatter) {
+              log.info(`Completed ${pkg.directory} (exit ${res.exitCode})`);
+            }
+            return res;
+          });
+        };
+      });
+
+    let results: EslintRunResult[] = [];
+
+    if (formatter) {
+      // When a formatter is set, we display a progress bar where we update on each completed package.
+      const total = thunks.length;
+
+      const render = (done: number, tot: number) => {
+        const pct = tot === 0 ? 100 : Math.floor((done / tot) * 100);
+
+        log.write(`\rProgress: ${done}/${tot} (${pct}%)`);
+      };
+
+      render(0, total);
+
+      results = await runBatchedPromises(thunks, usedConcurrency, (done, tot) => render(done, tot));
+
+      // Finalize progress with a newline
+      log.write('\n');
+    } else {
+      // No formatter: stream worker output as-is, no progress bar
+      results = await runBatchedPromises(thunks, usedConcurrency);
+    }
+
+    if (formatter) {
+      const formatterFn = formatter
+        ? (await loadFormatterModule(formatter, log)) || defaultFormatter
+        : defaultFormatter;
+
+      formatterFn(log, results);
+    }
+
+    // Exit status
+    if (formatter) {
+      const totalFindings = results.reduce((acc, r) => {
+        if (!r || !Array.isArray(r.report)) return acc;
+
+        const count = r.report.reduce((fAcc, f) => fAcc + (f.messages?.length || 0), 0);
+
+        return acc + count;
+      }, 0);
+
+      if (totalFindings > 0) {
+        process.exit(1);
+      }
+    } else {
+      const hadFailures = results.some((r) => (r.exitCode ?? 0) !== 0);
+
+      if (hadFailures) {
+        process.exit(1);
+      }
+    }
+  },
+  {
+    description: 'Run a custom eslint config on kibana packages',
+    flags: {
+      string: ['concurrency', 'formatter', 'eslint-config', 'filter'],
+      default: {
+        concurrency: 1,
+        'eslint-config': 'eslintrc.js',
```

Comment:
Since this is about running custom eslint rulesets, maybe having a default is not meaningful, especially if this file doesn't exist. This could be made a required argument, or a requried positional